### ISSUE TITLE
Sensitivity to line-spacing

### ISSUE THREAD

**Issue by acowley** [jcfk/org-sliced-images#4]
Thank you for the package! It addresses a real awkward part of using org. A question I have is if there is any hope of being able to use non-nil `line-spacing` alongside sliced images. I enjoy increased reading comfort with a slight bump to `line-spacing`, but this introduces gaps in sliced images.

**Comment by jcfk** [comment]
Thanks. Could you try setting `line-spacing` to nil and putting the desired value in `default-text-properties`, eg `(setq default-text-properties '(line-spacing 5))`?

**Comment by acowley** [comment]
Thank you for the suggestion! That has the interesting effect of increasing the spacing around newlines (e.g. paragraph breaks), but not between consecutive lines of wrapped text.

**Comment by jcfk** [comment]
Gotcha. What if you also used `wrap-prefix`:

```
(setq my-line-spacing 5)
(setq default-text-properties
      `(line-spacing my-line-spacing
        wrap-prefix (space . (:height (,(+ (default-font-height) my-line-spacing))
                              :width (0)))))
```

**Comment by acowley** [comment]
This is a little strange. Given I wanted a `line-spacing` of 0.2, I can use the approach you suggest here by using 0.2 for the `line-spacing` text property to affect newlines and giving `wrap-prefix` a `:height` property of 1.2. What this does to a test with a sliced image is insert a gap after the first row, then a smaller gap after the second row, and then no more gaps for the rest of the image.

**Comment by jcfk** [comment]
Very odd. To summarize, on your machine:

1. `(setq default-text-properties '(line-spacing 0.2))` gives the desired spacing between logical lines, no spacing between visual lines, and no gaps between slices.
2. `(setq default-text-properties '(line-spacing 0.2 wrap-prefix (space . (:height 1.2 :width 0))))` gives the desired spacing between logical lines and visual lines, and differently-sized gaps between the first three slices.

The latter is not the case on my machine (emacs 29.3, org 9.6.15); there are no gaps between slices.

What is the nature of the links you are using (ie their length and placement wrt other text)? Is the same true with `emacs -Q`? What  versions of emacs and org do you have?

**Comment by acowley** [comment]
I'm testing on `GNU Emacs 29.3 (build 1, aarch64-apple-darwin23.4.0, NS appkit-2487.50 Version 14.4.1 (Build 23E224))` and `Org mode version 9.6.26`. I'm going to work with the `default-text-properties` approach and figure out why it's not always working, as it's very close for me. Thank you for your help!

In changing those settings back and forth a lot while toggling inline images, I did encounter some changes to my test document. In particular, non-headline text following an image would sometimes lose newlines between the image and the text. When this happened, I saw this in `*Messages*`: `org-sliced-images-remove-inline-images: Wrong type argument: number-or-marker-p, nil
next-line: End of buffer [2 times]` 

And this in `*Warnings*`: `⛔ Warning (org-element-cache): org-element--cache: Unregistered buffer modifications detected (796 != 794). Resetting.`

I'm only seeing it when updating these settings we're talking about. If I can make a reproducer, I'll open another Issue.
